### DISCUSSION TITLE
ide工程文件转xmake

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#5043] [General]
目前xmake有生成ide工程文件的命令xmake project，是否考虑添加ide工程文件转xmake呢？或者自动探测执行目录下是否有ide工程文件然后提示转成xmake这种

**Comment by star-hengxing** [comment]
理论上可以实现，不就是写个 parser 读取然后 codegen，实际上工程量太大不可能做到。

**Reply by waruqi** [reply]
能实现，和实现可靠稳定，这两的工作量相差的不是一点点。

xmake 转其他可以，是因为 xmake 能够方便取到自身工程的信息。但是反过来转，除非对方 ide / cmake 开放获取信息的接口，比如让 cmake 自身去支持转 xmake 一样。你要想外面走 parser 自己解析搞，我肯定是没时间折腾的。

而且这个即使实现，也不需要塞进 xmake，你们自己有兴趣的可以自己单独开项目，自己去转，用 py 还是其他啥脚本随意搞。没必要让 xmake 去支持。
